### ISSUE TITLE
Final decision of sources suppression as a function of radius from the phase  center

### ISSUE THREAD

**Issue by atemkeng** [atemkeng/Windowing-Functions#10]
So, what is our final decision? Averaging 100s or 50s over time? Averaging 6.2MhZ or above over frequency at 1.4GHz. I have to produce these graphs. Also, what is the final field of view? 1deg and 2deg? 2deg and 3deg? or ...

@o-smirnov

**Comment by o-smirnov** [comment]
I can't say it's final yet, so let's make all the following combinations
and then refine based on what the results look like:

Filters: avg, sinc, sinc 100x25, sinc 150x50

FoVs: 1, 2, 4 deg

Bandwidth: 1 MHz, 2 MHz, 4 MHz

Time: 25s, 50s, 100s

Calculate suppression as a function of radius (r=0...6 deg)

I count 108 combinations in total, think it's reasonable?

Also, as discussed, I'd like to see: flux of source in image plane (natural
& uniform weighting), amplitude of longest baseline, amplitude of medium
baseline, amplitude of shortest baseline.

Thus for each of the 108 combinations you need to generate four vectors of
numbers. Store them in a cPickle or something, because I'd like to load
them into a notebook and play with the best way of presenting the plots.

**Comment by o-smirnov** [comment]
And besides those four vectors of numbers, we also need the noise ratios of course.

**Comment by atemkeng** [comment]
Yes ~ 108 combinations, in exact 90 combinations as Avg doesn't varies as a function of a parameter that defined the FoV. Thus,  9 combinations for Avg and 27 for each of the sinc. I will produce the plots.

**Comment by o-smirnov** [comment]
Yes, and get me the raw data that goes into the plots as well please.

**Comment by atemkeng** [comment]
To produce all these plots under 4hrs observation is a had journey! takes a lot of time to process. I wrote the pipe line and run some few (18plots) yesterday till now in process

**Comment by o-smirnov** [comment]
How long? If too long (i.e. more than a day), then just give me snapshots.
Remember, this is exploratory still, I'm still going to ask you for a
"final" set of plots based on what we find.

**Comment by atemkeng** [comment]
Ok cooll, I think I should do this for snapshots and when you find a  suitable plots let me know I should produce directly. The pipe line is 18 curves for 1Mhz, sinc-0x0 and sinc-100x25 for all FoV, all integrations as you mentioned. yesterday at 7pm till know on going.

**Comment by atemkeng** [comment]
I just noticed something on the thermal noise. It is very cool to re-sample over short time like (25s,50s) and use wide FoV (3deg, 4deg) and overlap the filters (sinc 3x2, sinc 4x3). The noise goes down well. I will send you some of the results.

**Comment by o-smirnov** [comment]
Good, I was just going to ask for a shorter timescale result. Do a 1deg fov
as well please.

**Comment by atemkeng** [comment]
Check out 30 plots data for 1Mhz and all integrations (25s,50s,100s) and all possible overlap. See the ReadMe.

https://github.com/atemkeng/Windowing-Functions/tree/master/ResultsData

**Comment by atemkeng** [comment]
And in this regime (25s, 50s) sinc(r) will do well.

**Comment by atemkeng** [comment]
Longest, Shortest and medium baseline length source suppression as a function of radius.

![baseline_longmidshort](https://cloud.githubusercontent.com/assets/6806881/8165081/bcc88a40-138d-11e5-9e32-38b0862980bd.png)

**Comment by o-smirnov** [comment]
Aha, looking good. But, I don't see the red curve. Also, the legend has
four "avg" and only two "sinc" entries?

**Comment by o-smirnov** [comment]
I can't load your pickle. Getting "ValueError: insecure string pickle". The interwebs suggests this may be due to you forgetting to close the file, so it wasn't completely written...

While we're at it, could you please include the per-baseline (shortest, medium, longest) amplitudes in the data structure? Your notes suggest this data is not in there yet.

Also, make sure you take shortest, medium, longest in terms of E-W distance on Earth, rather than absolute length.

**Comment by atemkeng** [comment]
Sorry, these are the update. 
![baseline_longmidshort](https://cloud.githubusercontent.com/assets/6806881/8165324/ccb2664a-138f-11e5-8ea9-2a67a8baf5a0.png)

Zoom on the Mid baselines Sinc Vs Avg

![baseline_longmidshortzommedium](https://cloud.githubusercontent.com/assets/6806881/8165334/e0fddc74-138f-11e5-8c06-57fffeaae66d.png)

Zoom on the shortest baseline Sinc Vs Avg

![baseline_longmidshortshortest](https://cloud.githubusercontent.com/assets/6806881/8165345/f9dec618-138f-11e5-8aa6-634df81ce13e.png)

**Comment by o-smirnov** [comment]
Ah I see. So even on the medium-length baselines the sinc is pretty much a boxcar. Is this for a Fov of 1deg? Non-overlap?

**Comment by atemkeng** [comment]
Yes,  a FoV of 1deg without overlap.

**Comment by o-smirnov** [comment]
OK, makes sense. Going to need a massive overlap for BDWFs to make a difference in this regime, otherwise we're just fiddling with longer baselines. Any reason why we can't do an overlap factor of 10? Of 100?

**Comment by atemkeng** [comment]
Yeah we can do this but not for a 1 deg FoV. For a FoV >1 it is interesting and more interesting when we re-sample over short time. As you mentioned in the paper, the computational time is the only drawback.

**Comment by o-smirnov** [comment]
Why can't we do this for a 1deg FoV?

**Comment by atemkeng** [comment]
When the overlap time and/or frequency get too large, we started to suppress across the Field of view, what is not good although be reduce the noise significantly by doing that.

**Comment by o-smirnov** [comment]
That's strange, why would that be? Need to understand this better.

**Comment by o-smirnov** [comment]
Interesting. At FoV=1deg, 1 MHz, there is absolutely no gain in any of the BDWFs as far as smearing is concerned, they all do worse than averaging. Even at FoV=2deg, the gains are miniscule. Your old plots seem to show BDWFs doing a lot better -- but they were at 6.25 MHz -- so let's wait for the higher-bandwidth plots.

![download 20](https://cloud.githubusercontent.com/assets/6470079/8170237/866ebc4a-13af-11e5-999a-798282024556.png)
![download 19](https://cloud.githubusercontent.com/assets/6470079/8170238/8693c80a-13af-11e5-9a96-43525d36b546.png)
![download 18](https://cloud.githubusercontent.com/assets/6470079/8170239/8699982a-13af-11e5-99ed-c74f2d0c25a2.png)

**Comment by o-smirnov** [comment]
By the way, is this sinc(r), or sinc(x)sinc(y)?

**Comment by atemkeng** [comment]
All are Sinc(x)sinc(y). But my mind tell me that sinc(r) will work better for this regime. I have to run  and see.

-------- Original message --------

**Comment by atemkeng** [comment]
I have added 2MHz data in the same Rep just the naming that change.

**Comment by o-smirnov** [comment]
OK check out the notebook at https://github.com/o-smirnov/Windowing-Functions/tree/master/paper1/data.

**Comment by o-smirnov** [comment]
2MHz looks "more useful". But see the calculation at the end of the notebook. In terms of "uv-distance" 100s is 10MHz at 1.4Ghz. We should've worked this out earlier!

Anyway what this tells me is that we also need data at 8 MHz BW to complete the picture. 100s/8MHz and 50s/4MHz and 25s/2MHz are "balanced" in that the sampling bins become square-ish. 100s/1MHz is not balanced at all (long "thin" bins), hence the poor results. Will be a useful point to make in the paper.

**Comment by atemkeng** [comment]
I prefer to make Figure 12. and table3 when we have chosen a proper window. I have made correction in the paper. Thanks

**Comment by atemkeng** [comment]
I have added 10Mhz and 100s,  2.5Mhz  and 25s.

 https://github.com/atemkeng/Windowing-Functions/blob/master/ResultsData/picklefile_chandwidth-10Mhz-int-100s_2.5Mhz-int-25s_FoV-1-2-4deg_overlaps.data

**Comment by atemkeng** [comment]
This is just a proof of concept. I have simulated a 4hrs hires of 1s integration then I  re-sampled to 100s and 1Mhz channels width to see how the noise increased. The noise doesn't rely increase you can compare the plot with the snapshot of 100s and 1Mhz. But I noticed that the noise is amplified when I increase the channels widths to 2Mhz, .....

![plot1mhz-4hrs](https://cloud.githubusercontent.com/assets/6806881/8289628/f6a03552-191d-11e5-8bcb-082518ce27f4.png)

The plot below is from a 4hrs observation, integration 1s and 180 channels of width 0.0669MHz. I then re-sampled the time to 100s (average 100 samples) and the channels to 1Mhz (average 15 samples) then applied sinc-4x3. Indeed the different between this plot and the above plot is on the number of channels of the lores MS. Here the lores MS has 10 channels and the above the lores MS has 1 Channel. see the data and the configuration file:

https://github.com/atemkeng/Windowing-Functions/blob/master/ResultsData/picklefile_chandwidth-1Mhz_int-100s_FoV-2deg_overlaps.data

https://github.com/atemkeng/Windowing-Functions/blob/master/ResultsData/wfconfigfileVLAC-Bandwidth-1MhZ_Sinc-150x50_FoV-2deg_int-100s.cfg

This illustrate what you mentioned the last time. The noise increases. 

![sinc-avg-4hrobser](https://cloud.githubusercontent.com/assets/6806881/8647579/81501280-2959-11e5-8e4e-9f7218e0416d.png)

@o-smirnov

**Comment by atemkeng** [comment]
Run configuration https://github.com/atemkeng/Windowing-Functions/blob/master/Code/wfconfigfileVLAC-Bandwidth-1MhZ_Sinc-150x50_FoV-2deg_int-100s.cfg

python runwf-fluxradius.py "ls wfconfigfileVLAC-Bandwidth-1MhZ_Sinc-150x50_FoV-2deg_int-100s.cfg"

**Comment by o-smirnov** [comment]
Your latest data update is missing the "avg" entry for int25s.

**Comment by atemkeng** [comment]
The avg entry is in, see below or maybe I don't understand well the
question.

In [14]: file = pickle.load( open(
"picklefile_chandwidth-10Mhz-int-100s__5MHz-int-50s_2.5Mhz-int-25s_FoV-1-2-4deg_overlaps.data",
"rb"))

In [15]: file.keys()
Out[15]: ['bandwidth2.5Mhz', 'bandwidth10Mhz', 'bandwidth5Mhz']

In [16]: file['bandwidth2.5Mhz']['int25s']['avg']
Out[16]:
{'flux': array([ 1.00000072,  0.99432814,  0.98798674,  0.98429561,
 0.97074866,
        0.94986469,  0.94238389,  0.92736363,  0.90549356,  0.88878238,
        0.87042594,  0.84695655,  0.81951201,  0.79293972,  0.77000087,
        0.74441427,  0.71704745,  0.68868494,  0.66035086,  0.63737839,
        0.61452162,  0.59138399,  0.56836116,  0.54577726,  0.52386862,
        0.50282097,  0.48292813,  0.46506119,  0.44789526,  0.43149257,
        0.41588369,  0.40106604,  0.38703731,  0.3737621 ,  0.3612107 ,
        0.34933251,  0.3380923 ]),
 'noise': 0.00033000507}

**Comment by o-smirnov** [comment]
It's not there for 10MHz/25s though.

Please also provide 10MHz/50s for completeness.

**Comment by atemkeng** [comment]
Ok I get it now. The data is only for 2.5MHz/25s, 5MHz/50s and 10MHz/100s.

**Comment by o-smirnov** [comment]
I'm still bothered by the sinc2 results going _above_ unity in recovered flux. As far as I can see, this is mathematically impossible. For a 1 Jy source off-centre, all the input (hi-res) visibilities have unity amplitude and some phase. The sum of N complex numbers of unity amplitude cannot have an amplitude larger than N. It has to be an error in your normalization somehow! Please triple-check.

**Comment by atemkeng** [comment]
10MHz and 50s, 10MHz and 25s added. See link below 

https://github.com/atemkeng/Windowing-Functions/blob/master/ResultsData/picklefile_chandwidth-10Mhz-int-50s_ints-25s_FoV-1-2-4deg_overlaps.data

**Comment by atemkeng** [comment]
Looks very strange to me. I have to check this with the hi-res data and see if I have the same fluctuation within a radius of 2deg.

Observed frequency 14GHz
channels width 10Mhz
integration time 10s

![sinc-avg](https://cloud.githubusercontent.com/assets/6806881/8594914/00a2b2d4-2646-11e5-95a7-b93aa432a73b.png)

Need to increase the resolution upto 0.25arcsec then you will the following

![sinc-avg](https://cloud.githubusercontent.com/assets/6806881/8624143/1cd5f98e-2736-11e5-960e-6a90509348b8.png)

@o-smirnov

**Comment by atemkeng** [comment]
Visibility amplitude Vs radius @o-smirnov 

![figure_1](https://cloud.githubusercontent.com/assets/6806881/8618010/0e385c40-270a-11e5-8cad-045f03a53f56.png)

**Comment by atemkeng** [comment]
A 5s re-sampling time and 16Mhz re-sampling frequency plot. Here the noise is kite amplify with 2deg sinc, the noise ratio is 6.412. It is just very difficult to find where averaging drop below 0.9 in this regime (5s, 16Mhz). I am simulating again with a filter of 0.5deg, 1deg to see how the noise behave and also with Bessel. It is clear that with BDWF's the is no smearing  over the VLBI FoV but now as you said the noise matter, I have to fine the best filter for this. @o-smirnov 

![int5s16mhz](https://cloud.githubusercontent.com/assets/6806881/9033719/271cf37e-39ca-11e5-9da8-498a9dd53789.png)

**Comment by o-smirnov** [comment]
Let's split VLBI out into a separate issue.

In the meantime, I spent the flight trying to get the last section of the paper into shape. Here'sthe "final" plots I've arrived at:

![download 26](https://cloud.githubusercontent.com/assets/6470079/9058644/0915429e-3aa3-11e5-984f-0501e92df7b2.png)

![download 27](https://cloud.githubusercontent.com/assets/6470079/9058645/09179616-3aa3-11e5-841e-ecd7591baefa.png)

I think these are the filters we'll want to do a full run for. But first you must figure out the noise situation. I have looked at the noise obtained by different averaging settings:

```
(10.0, 100, 'avg', None) 0.000311286
(2.0, 25, 'avg', None) 0.000330005
(4.0, 100, 'avg', None) 0.000478045
(1.0, 100, 'avg', None) 0.000487739
(1.0, 50, 'avg', None) 0.000451033
(4.0, 25, 'avg', None) 0.000505531
(1.0, 25, 'avg', None) 0.000471339
(4.0, 50, 'avg', None) 0.000461966
(2.0, 100, 'avg', None) 0.000497547
(2.0, 50, 'avg', None) 0.000491046
```

This makes no sense! For boxcar, noise should be the same no matter how you average, since in the end the 0,0 pixel of the image is just a sum of all the high-res visibilities. The discrepancy worries me.

**Comment by atemkeng** [comment]
Yes but this is not how I am computing the noise. I have to make a noisy image and take the std. I wonder if the noise will stay constant  for the same resolution and image size. Anyway I will check this again.

**Comment by o-smirnov** [comment]
Yeah there will be subtle differences. Your pixels have to sample the psf
by a factor of several. Check the two "low noise" cases first.

**Comment by atemkeng** [comment]
I verified ,
for  (10.0, 100, 'avg', None) I simulated to time and I got the following noise 0.00030597381 and 0.00030580908.

**Comment by o-smirnov** [comment]
Compare (1.0, 100, 'avg', None) and (10.0, 100, 'avg', None). You should have the same (or at least very close) image noise, as long as the number of output channels in the second case is 1/10th of those in the first case (i.e. the effective output frequency range is the same).

**Comment by atemkeng** [comment]
I really do not know what happen. I run all the configurations files again.

For (1., 100, 'avg', None) I simulated this twice to see, and this is what I have 0.00031786656 and 0.00031524149. The hires, lores MS and the configuration file are here: elwood: /home/atemkeng/NVSS/TEST-VERIFICATIONXXXXXX/data-1MHz-100s

For (10., 100, 'avg', None) I also simulated this twice, and this is what I have 0.00032379036 and 0.00031327232. The hires, lores MS and the configuration file are here: elwood: /home/atemkeng/NVSS/TEST-VERIFICATIONXXXXXX/data-10MHz-100s

@o-smirnov

**Comment by o-smirnov** [comment]
OK, I can only assume you changed or fixed something in your scripts midway through. Please understand the problem. We're going to need to re-run everything from scratch now. May as well use the new intervals we settled on: 25s, 50s, 100s, 2.5 MHz, 5 MHz, 10 MHz. 

Start with the combinations 25x2.5, 50x5, 100x10, and the filters I used in my plots above. I can then at least re-do the plots and carry on with the interpretation.

**Comment by atemkeng** [comment]
Refer to https://github.com/atemkeng/Windowing-Functions/tree/master/FinalData
for the data, 25x2.5, 50x5, 100x10 for sinc-1x1,sinc-4x3,bessel-1x1,bessel-4x3 and averaging and the configurations files that you can also run.

Notice that I have change the hires channels width to be able to do this as with the old hires channels width I couldn't find  "number-of-resample-channels" such that {channel-width}x{number-of-resample-channels}=2.5MHz.

I will suggest that in your above plots we should use for averaging: (25s, 2.5MHz), (50s, 5MHz) instead of (25s, 2MHz) and (50s, 4MHz) to keep the configurations parameters the same and avoid the noise problem. 

@o-smirnov

**Comment by o-smirnov** [comment]
Yep, I'll switch to 25/2.5 and 50/5 for consistency throughout. Thanks,
I'll redo the plots tonight and see what things look like.

**Comment by o-smirnov** [comment]
The 2.5/25 data file is incomplete, cPickle can't load it. Looks like you didn't save it properly! Please redo.

**Comment by o-smirnov** [comment]
OK, these are the new plots:

![download 28](https://cloud.githubusercontent.com/assets/6470079/9220381/885a12c4-40e2-11e5-8dbb-0cb56911ff4b.png)

![download 29](https://cloud.githubusercontent.com/assets/6470079/9220382/885cf3a4-40e2-11e5-8db9-62896f96b04a.png)

Now need a 4-hour synthesis with 20 output channels (100s, 10MHz -> 4h, 200MHz) for {sinc,bessel}-4x3-{1,2}deg. NB: the hi-res MS will need to be bit wider to accommodate overlap.

**Comment by atemkeng** [comment]
I sampled a hires at each 1s during 35min (here 5 minutes allows me to overlap 2.5min at the right and 2.5min at left). And I sampled 220 channels of 1MHz each, resulting to a total bandwidth of 220.0MHz (20MHz allows me to overlap 10MHz at the left and 10MHz at the right ). I then re-sampled the hires during 100s and 10MHz. The results are here: https://github.com/atemkeng/Windowing-Functions/blob/master/FinalData/channels_lores_20-timeslot_lores_18-sinc-bessel-bw10MHz-int100s.data

@o-smirnov

**Comment by o-smirnov** [comment]
OK, still have a noise "penalty" of 0.79. I don't see how this is possible!
In fact it seems mathematically impossible.

Can you please point me to your code and config file used for this run?

**Comment by o-smirnov** [comment]
Could you please point me to the high-res MS, and the low-res MSs, and the
scripts and logs from your run? I would like to recreate this exactly.

**Comment by atemkeng** [comment]
I had make things easy to analyse just the noises. I reduced this script just for the noise so that you could  see things better. See elwood: /home/atemkeng/NVSS4HRSOBSERVEDFINAL/New-REP

run: pyxis noise_estimate[hiresms="hires-35s-220MHz.MS",lowresms="lores-300min-200MHz.MS"]

The output will save you a pickle file for the noise by default with name  "pickle-noise.data" or take the name in parameter as 
 pyxis noise_estimate[hiresms="hires-35s-220MHz.MS",lowresms="lores-300min-200MHz.MS", filename = "xxx.data"]

open the scripts pyxis-run-noise.py and you will see a parameter for FoVs, overlaps.

I will prefer that you should just run this as it will not take a lot of time.

Let us have a skype please, so that I can explained you what I understand for a 1deg FoV window and some plots. To me nothing is strange for the noise ratio of overlap 1deg window been less than 1. You mentioned this earlier during the last busy week in Grahamstown when Cyril were there and I verified.

**Comment by atemkeng** [comment]
Everything is correct. I did this long time  ago for 4 hours and observed the same. This is a second run as previously mentioned. See output

{'bessel-4x3_1FoV': 0.00021295228,
 'bessel-4x3_2FoV': 0.00034026237,
 'bessel-4x3_4FoV': 0.00064905547,
 'boxcar-100dtime-10dfreq': 0.00027728296,
 'sinc-4x3_1FoV': 0.00021486981,
 'sinc-4x3_2FoV': 0.00035773678,
 'sinc-4x3_4FoV': 0.00073335023}

**Comment by o-smirnov** [comment]
OK bad news is, I found the catch. There had to be a catch, I told you, those numbers were too good to be true. :) Good news is, the real numbers are pretty decent as well:

'boxcar-100dtime-10dfreq': 6.4834632e-05,
'sinc-4x3_1FoV': 6.2136831e-05, 
'bessel-4x3_1FoV': 6.22019e-05,
'sinc-4x3_2FoV': 8.6298714e-05
'bessel-4x3_2FoV': 8.0588063e-05,
'sinc-4x3_4FoV': 0.00017699273, 
'bessel-4x3_4FoV': 0.00015719925, 

Here's where the problem was: you don't specify a channel range for the imager. You need to either explicitly set `ms.CHANRANGE=0,19`, or set `v.MS = msname` (to set it up automatically based on the content of the MS). Without it, it was running lwimager with img_chanstep=0 which (I think) caused lwimager to use only one channel rather than all 20. It's a bit of funny behaviour (arguably an lwimager bug), but this is what confused your results.

If you only use one channel, overlapping BDWFs perform much better, since they're "borrowing" signal from neighbouring channels. 

The numbers above are far more believable. Note that the noise of the FoV=1 overlapping BDWFs is still lower than boxcar, but only by a few percent. This is again because of the "borrowing" effect, and because the simulation is small. The boxcar is adding up signal from 18 timeslots and 20 channels, the BDWFs are adding up signal from 21 timeslots and 22 channels. sqrt(21_22/(18_20)) = 1.13, i.e. the BDWFs have a 13% extra "sensitivity budget" because of the extra signal from the overlaps at the edges. Of course they can only reach the full 13% if the BDWF is identical to an overlapping boxcar, so the fact that they don't is as expected.

If you do a larger simulation say (4 hour, 200 channel), the extra overlaps at the edges are going to be much smaller relative to the data available, so I predict the noise from FoV=1 BDWFs will then become slightly higher than boxcar, as expected all along.

**Comment by atemkeng** [comment]
Data at 14GHz observed frequency (see link below)
I sampled a hires at each 1s during 30min30s  (here 30s  allows me to overlap 15s at the right and 15s at left). And I sampled 220 channels of 1MHz each, resulting to a total bandwidth of 220.0MHz (20MHz allows me to overlap 10MHz at the left and 10MHz at the right ). I then re-sampled the hires during 10s and 10MHz.

https://github.com/atemkeng/Windowing-Functions/blob/master/FinalData/data-14GHz-Observation/channels_lores_20-timeslot_lores_18-sinc-bessel-observed-frequency-14Ghz-bw10MHz-int100s.data

You were just asking the data for all combinations. This is going to take more time as the Hires is about 220 channels regularly sampled at each 1s during 30min30s . It is really necessary? As we showed these already for snapshots. Anyway,  things are still running.
